executors/sshexec/block_volume.go

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright (c) 2017 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package sshexec
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/heketi/heketi/executors"
+	"github.com/lpabon/godbc"
+)
+
+func (s *SshExecutor) BlockVolumeCreate(host string,
+	volume *executors.BlockVolumeRequest) (*executors.BlockVolumeInfo, error) {
+
+	godbc.Require(volume != nil)
+	godbc.Require(host != "")
+	godbc.Require(volume.Name != "")
+
+	type CliOutput struct {
+		Iqn      string   `json:"IQN"`
+		Username string   `json:"USERNAME"`
+		Password string   `json:"PASSWORD"`
+		Portal   []string `json:"PORTAL(S)"`
+		Result   string   `json:"RESULT"`
+		ErrCode  int      `json:"errCode"`
+		ErrMsg   string   `json:"errMsg"`
+	}
+
+	var auth_set string
+	if volume.Auth {
+		auth_set = "enable"
+	} else {
+		auth_set = "disable"
+	}
+
+	cmd := fmt.Sprintf("gluster-block create %v/%v  ha %v auth %v  %v %vG --json",
+		volume.GlusterVolumeName, volume.Name, volume.Hacount, auth_set, strings.Join(volume.BlockHosts, ","), volume.Size)
```

Comment:
@MohamedAshiqrh we also need to enable the gluster-block-profile on the hosting FS volume at this time.

Reply:
@obnoxxx `gluster volume set <vol1> group gluster-block and restart the volume` is required for each block hosting volume.

Reply:
thanks, @humblec will work this in